src/nbl/ext/ImGui/ImGui.cpp

```diff
@@ -681,116 +896,68 @@ namespace nbl::ext::imgui
 				.minDepth = 0.0f,
 				.maxDepth = 1.0f,
 			};
-			commandBuffer->setViewport(0, 1, &viewport);
 
-			// Setup scale and translation:
-			// Our visible imgui space lies from draw_data->DisplayPps (top left) to draw_data->DisplayPos+data_data->DisplaySize (bottom right). DisplayPos is (0,0) for single viewport apps.
+			commandBuffer->setViewport(0, 1, &viewport);
 			{
-				PushConstants constants{};
-				constants.scale[0] = 2.0f / drawData->DisplaySize.x;
-				constants.scale[1] = 2.0f / drawData->DisplaySize.y;
-				constants.translate[0] = -1.0f - drawData->DisplayPos.x * constants.scale[0];
-				constants.translate[1] = -1.0f - drawData->DisplayPos.y * constants.scale[1];
-
-				commandBuffer->pushConstants(pipeline->getLayout(), IShader::ESS_VERTEX, 0, sizeof(constants), &constants);
+				VkRect2D scissor[] = { {.offset = {(int32_t)viewport.x, (int32_t)viewport.y}, .extent = {(uint32_t)viewport.width, (uint32_t)viewport.height}} };
+				commandBuffer->setScissor(scissor); // cover whole viewport (only to not throw validation errors)
```

Comment:
just create a pipeline without any scissors (its in the pipeline creation parameters)

Reply:
you cannot because currently we hardcode dynamic state - [VK_DYNAMIC_STATE_SCISSOR](https://github.com/Devsh-Graphics-Programming/Nabla/blob/a10ab3d0ce1626a1d5ea8ed7d2e8a0aca4665462/src/nbl/video/CVulkanLogicalDevice.cpp#L1210) as well

Reply:
ask user for the number of dynamic scissors that will be used for the Graphics Pipeline

Reply:
where do you ask for the number of dynamic scissors?

Reply:
> where do you ask for the number of dynamic scissors?

I get a span of scissors from the [call args](https://github.com/Devsh-Graphics-Programming/Nabla/blob/gpgpuimgui/src/nbl/ext/ImGui/ImGui.cpp#L893)

Reply:
but that span's length needs to match the viewport count in the pipeline creation, and you never tell the pipeline how many scissors&viewports it will have at creation time

Reply:
resolved